[jxunze] iP

jxunze opened the pull request:
# Iris

> In every 60 seconds, a minute passes. 

Introducing _Iris_, your personal assistant to remind you of your tasks. It aims to help you:

  * [x]   Be more organised
  * [ ]   Be less lonely 😪


## Features

_Iris_ currently has the following features:

  * Todos
  * Deadlines
  * Events

### Get Started
1. Download the `.jar` file from [here](https://github.com/jxunze/ip/releases/tag/A-Jar).
2. Run the following command in the directory where the `.jar` file is located at:
```java
java -jar iris.jar
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jxunze committed:
Add skeleton

jxunze committed:
Add echo functionality

jxunze committed:
Add list functionality

jxunze committed:
add to-do functionality

jxunze committed:
Add different task types

jxunze committed:
Add tests

jxunze committed:
Add error messages

jxunze committed:
Add delete functionality

jxunze committed:
Add cache functionality

jxunze committed:
Merge branch 'branch-Level-7'
Merge branch-7 into master

jxunze committed:
Add datetime functionality

jxunze committed:
Add datetime functionality

jxunze committed:
Edit test cases

jxunze committed:
Merge branch 'branch-Level-8'
Merge with Level-8

jxunze committed:
Refactor classes

jxunze committed:
Merge branch 'A-MoreOOP'
Add OOP

jxunze committed:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle
Add gradle support

jxunze committed:
Gradle config added

jxunze committed:
Add tests for TaskList and StringUtils

jxunze committed:
Merge branch 'A-Gradle'
Add Gradle and tests

jxunze committed:
Add jar files

jxunze committed:
Merge branch 'A-Jar'
Add Jar

jxunze committed:
Add Javadocs

jxunze committed:
Change method name in Storage

jxunze committed:
Add find command

jxunze committed:
Merge branch 'A-JavaDoc'
Add JavaDocs

jxunze committed:
Resolve merge conflicts

jxunze committed:
Merge branch 'Level-9'
Add find functionality

whelan-low commented on src/main/java/Commands/DeleteCommand.java:
```diff
@@ -0,0 +1,40 @@
+    public String execute() throws IOException {
+        Task toBeDeleted = this.taskList.getTask(this.index);
+        this.taskList.delete(this.index);
+        String outputString = "I have removed the following task:\n" + toBeDeleted.toString()+"\nNow you have "+ this.taskList.size() + " tasks in your list.";
```
This line is much longer than 120 characters. Please break it down into multiple lines

whelan-low commented on src/main/java/Parser/Parser.java:
```diff
@@ -0,0 +1,151 @@
+            }
+        } catch (Exception e){
+            return new InvalidCommand("Invalid input for deadline. Input your deadline as such:\ndeadline <name_of_deadline> /by <due_date>");
+        }
```
Lines 131 and 134, reminder to trim them below 120 characters

whelan-low commented on src/main/java/Parser/Parser.java:
```diff
@@ -0,0 +1,151 @@
+                String eventName = matcher.group(1);
+                return new AddEventCommand(eventName, parseDateTime(matcher.group(2).trim()), parseDateTime(matcher.group(3).trim()));
+            } else {
+                return new InvalidCommand("Invalid input for event. Input your event as such:\nevent <name_of_event> /from <start_time> /to <end_time>");
```
Lines 112 and 114: trim line to below 120 characters

whelan-low commented on src/main/java/Storage/Storage.java:
```diff
@@ -0,0 +1,70 @@
+            Parser parser = new Parser();
+            while (s.hasNext()) {
+                String line = s.nextLine();
+                String[] str = line.split(" ");
```
a name better than str might be better for readability, perhaps words?

whelan-low commented on src/main/java/TaskList/TaskList.java:
```diff
@@ -0,0 +1,39 @@
+    private List<Task> getTasks() {
+        return this.cachedTasks;
+    }
+    public void clear() { this.cachedTasks.clear();}
```
Egyptian style braces to be used for clear() and size()

whelan-low commented on src/main/java/UI/IrisUI.java:
```diff
@@ -0,0 +1,57 @@
+    public void showInitFailedMessage() {
+        this.printWithDelimiter("Unable to read cache");
+    }
+}
```
Javadocs not done for this class

whelan-low replied:
Javadocs not done for this class

whelan-low reviewed (commented):
Overall, good structure and style of program! Good usage of the ` Command ` class. Most errors were lines that went beyond 120 characters.

jxunze committed:
Implement GUI

jxunze committed:
Prepare for merge

jxunze committed:
Add assertions

jxunze committed:
Improve code quality

jxunze committed:
Merge pull request #1 from jxunze/A-Assertions

Add assertions

jxunze committed:
Merge pull request #2 from jxunze/A-CodeQuality

Improve code quality

jxunze committed:
remove unused files

jxunze committed:
Add folder storage

Allow users to switch between folders in the within the application so
that the user can have separate repositories for different categories.
As such, 3 new commands are added:
* ls (list directories)
* mkdir (make repository)
* cd (change repository)

Potential vulnerability might be the case of injects, inserting and
manipulating files in the project folder.

jxunze committed:
Merge branch 'BCD-Extension'

jxunze committed:
Add user guide

jxunze committed:
add UI
